task-maker-async/src/worker.rs

```diff
@@ -0,0 +1,566 @@
+            }
+            ExecutionFileMode::Fifo(fifo) => {
+                hard_link(&group_dir.join(FIFO_DIR_NAME).join(fifo), local_path).await?;
+            }
+            ExecutionFileMode::Output => {}
+        };
+        Ok(())
+    }))
+    .await?;
+
+    // Write /etc/passwd in the sandbox root, as some software expects it to be present.
+    // TODO(veluca): this also ought to be done by tabox.
+    let etc = box_path(&dir, &ExecutionPath::Path(PathBuf::new().join("etc")));
+    create_dir_all(&etc).await?;
+    tokio::fs::write(
+        etc.join("passwd"),
+        "root::0:0::/:/bin/sh\nnobody::1000:1000::/:/bin/sh\n",
+    )
+    .await?;
+
+    let locked_path_for_file = path_for_file.lock().unwrap();
+    let sandbox_config =
+        prepare_sandbox_config(execution.clone(), dir.clone(), &locked_path_for_file)?;
+
+    Ok(ExecutionInfo {
+        sandbox_config,
+        execution,
+        dir,
+    })
+}
+
+struct ExecutionResultInfo {
+    dir: PathBuf,
+    execution: Execution,
+    result: ExecutionResult,
+}
+
+pub fn status_and_resources(
+    sandbox_status: &SandboxExecutionResult,
+    limits: &ExecutionLimits,
+) -> (ExecutionStatus, ExecutionResourcesUsage) {
+    let resources = ExecutionResourcesUsage {
+        cpu_time: sandbox_status.resource_usage.user_cpu_time,
+        sys_time: sandbox_status.resource_usage.system_cpu_time,
+        wall_time: sandbox_status.resource_usage.wall_time_usage,
+        memory: sandbox_status.resource_usage.memory_usage / 1024,
+    };
+
+    // it's important to check those before the signals because exceeding those
+    // limits may trigger a SIGKILL from the sandbox
+    if let Some(cpu_time_limit) = limits.cpu_time {
+        if resources.cpu_time > cpu_time_limit.as_secs_f64() {
+            return (ExecutionStatus::TimeLimitExceeded, resources);
+        }
+    }
+    if let Some(sys_time_limit) = limits.sys_time {
+        if resources.sys_time > sys_time_limit.as_secs_f64() {
+            return (ExecutionStatus::SysTimeLimitExceeded, resources);
+        }
+    }
+    if let Some(wall_time_limit) = limits.wall_time {
+        if resources.wall_time > wall_time_limit.as_secs_f64() {
+            return (ExecutionStatus::WallTimeLimitExceeded, resources);
+        }
+    }
+    if let Some(memory_limit) = limits.memory {
+        if resources.memory > memory_limit {
+            return (ExecutionStatus::MemoryLimitExceeded, resources);
+        }
+    }
+    match sandbox_status.status {
+        ExitStatus::Signal(signal) => (
+            ExecutionStatus::Signal(
+                signal as u32,
+                sandbox_status
+                    .status
+                    .signal_name()
+                    .unwrap_or_else(|| "unknown signal".into()),
+            ),
+            resources,
+        ),
+        ExitStatus::ExitCode(0) => (ExecutionStatus::Success, resources),
+        ExitStatus::Killed => (ExecutionStatus::WallTimeLimitExceeded, resources),
```

Comment:
I think the old behavior was: https://github.com/edomora97/task-maker-rust/blob/2095b4625bc23c28f3d27ec9c1892d94159a290b/task-maker-exec/src/sandbox.rs#L191